aspnetcore/blazor/components/render-modes.md

```diff
@@ -579,6 +516,228 @@ The following component results in a runtime error when the component is rendere
+
+In both cases, the component that must adopt static SSR must also force a full-page reload.
+
+The following examples use <xref:Microsoft.AspNetCore.Http.HttpContext>, which is available in statically-rendered root components, such as the `App` component (`App.razor`). When a non-root component is rendered, <xref:Microsoft.AspNetCore.Http.HttpContext> returns `null`, which is useful as a signal in app code to trigger a full-page reload.
+
+### Area (folder) of static SSR components
+
+The approach described in this subsection is used by the Blazor Web App project template with individual authentication and global interactivity.
+
+An area (folder) of the app contains the components that must adopt static SSR and only run on the server. For example, the Identity Razor components of the Blazor Web App project template are in the `Components/Account/Pages` folder.
+
+The folder also contains an `_Imports.razor` file, which applies a custom account layout to the components in the folder:
+
+```razor
+@using BlazorSample.Components.Account.Shared
+@layout AccountLayout
+```
+
+The `Shared` folder maintains the `AccountLayout` layout component. The component makes use of <xref:Microsoft.AspNetCore.Http.HttpContext> to determine if the component is rendering on the server. Identity components must render on the server with static SSR because they set Identity cookies. If the value of <xref:Microsoft.AspNetCore.Http.HttpContext> is `null`, the component is rendering interactively, and a full-page reload is performed by calling <xref:Microsoft.AspNetCore.Components.NavigationManager.Refresh%2A?displayProperty=nameWithType> with `forceLoad` set to `true`. This triggers a request to the server for the component.
+
+`Components/Account/Shared/AccountLayout.razor`:
+
+```razor
+@inherits LayoutComponentBase
+@layout BlazorSample.Components.Layout.MainLayout
+@inject NavigationManager NavigationManager
+
+@if (HttpContext is null)
+{
+    <p>Loading...</p>
+}
+else
+{
+    @Body
+}
+
+@code {
+    [CascadingParameter]
+    private HttpContext? HttpContext { get; set; }
+
+    protected override void OnParametersSet()
+    {
+        if (HttpContext is null)
+        {
+            NavigationManager.Refresh(forceReload: true);
+        }
+    }
+}
+```
+
+> [!NOTE]
+> In the Blazor Web App project template, there's a second layout file (`ManageLayout.razor` in the `Components/Account/Shared` folder) for Identity components in the `Components/Account/Pages/Manage` folder. The `Manage` folder has its own `_Imports.razor` file to apply to the `ManageLayout` to components in the folder. In your own apps, using nested `_Imports.razor` files is a useful approach for applying custom layouts to groups of pages.
+
+In the `App` component, any request for a component in the `Account` folder applies a `null` render mode, which enforces static SSR. Other component requests receive a global application of the interactive SSR render mode (`InteractiveServer`).
```

Comment:
> In the `App` component, any request for a component in the `Account` folder applies a `null` render mode, which enforces static SSR.

Just a small note about this: A `null` render mode is effectively the same as not specifying one at all, which results in the component inheriting its parent's render mode. In this case, the `App.razor` component gets rendered using static server rendering, so a `null` render mode makes the `<Routes>` component also use static server rendering. Whereas, if a `null` render mode were to be specified for a child component whose parent was interactive, the child would inherit the same interactive render mode.

All that to say, it's not always the case that `null` enforces static SSR, it just happens to behave that way in this example. Not sure whether it's worth calling out that caveat (maybe it would just create more confusion), so I'll leave it up to you to decide! Just want to make sure that readers don't see this statement and think that they can apply `null` as a render mode _anywhere_ to switch from interactive rendering to static server rendering.

Reply:
Yes ... great idea to add this. I placed it in an IMPORTANT note immediately after the paragraph with the remark. This is a *teachable moment*, to borrow an educational slogan, on this because we have the reader's focus on how this code is working and we're right in the thick of the whole render mode explanation in the article. Great idea to call it out.